source/geometry_model/spherical_shell.cc

```diff
@@ -76,6 +79,39 @@ namespace aspect
           Assert (false, ExcInternalError());
         }
 
+#if ((DEAL_II_VERSION_MAJOR>=9) ||        \
+     ((DEAL_II_VERSION_MAJOR==8) && (DEAL_II_VERSION_MINOR >= 2)))
+
+      // if deal.II is sufficiently new, then use a manifold
+      // description for all cells. use manifold_id 99 in order not to
+      // step on the boundary indicators used below
+      static const SphericalManifold<dim> spherical_manifold;
+      coarse_grid.set_manifold (99, spherical_manifold);
+
+      for (typename Triangulation<dim>::active_cell_iterator
+           cell = coarse_grid.begin_active();
+           cell != coarse_grid.end(); ++cell)
+        cell->set_all_manifold_ids (99);
+
+      // clear the manifold id from objects for which we have boundary
+      // objects (and need boundary objects because at the time of
+      // writing, only boundary objects provide normal vectors)
+      for (typename Triangulation<dim>::active_cell_iterator
+           cell = coarse_grid.begin_active();
+           cell != coarse_grid.end(); ++cell)
+        for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+          if (cell->at_boundary(f))
+            cell->face(f)->set_all_manifold_ids (numbers::invalid_manifold_id);
+
+      static const StraightBoundary<dim> straight_border;
+      std::set<types::boundary_id> ids = get_used_boundary_indicators();
+      for (std::set<types::boundary_id>::iterator it = ids.begin();
+           it!=ids.end(); ++it)
+        if (*it > 1)
+          coarse_grid.set_boundary (*it, straight_border);
```

Comment:
This is the place where I think a comment on why you're doing that (and which parts of the boundary it affects) is in place. I believe the word "border" only applies to the edge of a 2d object. Maybe call it `straight_boundary`?